troubleshooting updates and images video (#826)
chainguard-dev/edu: content/chainguard/chainguard-enforce/troubleshooting-tips.md

This page contains tips for troubleshooting problems that one may encounter when working with Chainguard Enforce.

## Installation

If you're encountering an issue with installation, we have a few recommendations to check, and guidance on using Rancher.

### Verify cluster is installed

> The `remoteId` in `chainctl cluster install` is the UUID of the Gulfstream namespace; this will help identify if a cluster is installed or not. You can learn more about installation in our [Installing the Chainguard Enforce Agent](/chainguard/chainguard-enforce/installation/alternative-installation-methods/) tutoria.

Reviewer: ```suggestion
The `remoteId` in `chainctl cluster install` is the UUID of the Gulfstream namespace; this will help identify whether a cluster is installed or not. You can learn more about installation in our [Installing the Chainguard Enforce Agent](/chainguard/chainguard-enforce/installation/alternative-installation-methods/) tutorial.
```

You can get the ID for Gulfstream with `kubectl`, as in the next command.

```sh
kubectl get namespace gulfstream -o json | jq .metadata.uid
```

You'll receive the ID.

```
"B501db05-3702-4403-81b5-c74a55976597"
```

With `chainctl cluster list`, you can also get this ID by passing the group ID.

```sh
chainctl cluster list --group $GROUP_ID -o json | jq -r .items[0].remoteId
```

The output here will also be the remote ID. 

```
b501db05-3702-4403-81b5-c74a55976597
```

You can review more options in the [`chainctl` docs](/chainguard/chainctl/chainctl-docs/chainctl_clusters_list/).

### Cluster OIDC configuration exit status

You may encounter an exit status due to being unable to fetch the OIDC configuration. This can occur when running the `chainctl cluster install` command.

```sh
chainctl cluster install --group=$GROUP --private --context demo 
```

> The error will be similar to the below.

Reviewer: ```suggestion
The error will be similar to the following.
```

```
...
Installing Chainguard agent...
couldn't fetch cluster OIDC configuration exit status 1
```

The cause of this may be due to the `openid-configuration` and `jwks` being inaccessible within the Kubernetes API. You can determine this with `kubectl`.

First, check whether the issue is with the `openid-configuration`.

```sh
kubectl get --raw /.well-known/openid-configuration
```

If this is the issue, you'll receive an error as output.

```
Error from server (NotFound): the server could not find the requested resource
```

Next, you can check if the issue is related to `jwks`.

```sh
kubectl get --raw /openid/v1/jwks
```

If `jwks` being inaccessible is the issue, you'll receive an error here.

```
Error from server (NotFound): the server could not find the requested resource
```

> The output above reveals that the endpoints are not available to `chainctl`, whcih may be due to an auth proxy between `chainctl` and the controlplane.

Reviewer: ```suggestion
This output reveals that the endpoints are not available to `chainctl`, whcih may be due to an auth proxy between `chainctl` and the control plane.
```

```sh
kubectl get --raw /.well-known/openid-configuration
```

If you receive an error that the requested resource was not found, make sure you are using the `--private` option on installation. You can review guidance on this option in the [Configuring Enforcer Options](/chainguard/chainguard-enforce/installation/configure-enforcer-options-during-installation/#install-with-chainctl) guide.

If you are using Rancher, you'll be able to generate “direct access” kubeconfigs that bypass the auth proxy and go directly to the cluster. Please follow the [Rancher documentation to create a new kubeconfig file](https://ranchermanager.docs.rancher.com/how-to-guides/new-user-guides/manage-clusters/access-clusters/use-kubectl-and-kubeconfig#authenticating-directly-with-a-downstream-cluster).

### Installing Enforce with Rancher

If you're using Rancher for Kubernetes, you'll need to point directly to your kubeconfig file:

```sh
KUBECONFIG=/etc/rancher/rke2/rke2.yaml chainctl cluster install --group=$GROUP --private
```

> If you cannot reach the OIDC endpoint, this will cause issues for a cluster install, review the [section above](#cluster-oidc-configuration-exit-status) for guidance on troubleshooting this. 

Reviewer: ```suggestion
If you cannot reach the OIDC endpoint, this will cause issues for a cluster install, review the [previous section](#cluster-oidc-configuration-exit-status) for guidance on troubleshooting this. 
```

Additionally, you'll need direct access to the authorized endpoint and not the Rancher proxy URL. When you run `kubectl config view`, you should receive something like the output below, without the Rancher URL.

```
clusters:
- cluster:
    server: <direct-url-to-cluster>
```

## Issues logging in with chainctl

Ensure that you're pointing to production. You can check the `chainctl` config.

```sh
chainctl config view 
```

The config should be pointing to the production environment, as in the next output.

```
platform:
    api: https://console-api.enforce.dev
    audience: https://console-api.enforce.dev
    console: https://console.enforce.dev
    issuer: https://issuer.enforce.dev
    registry: https://cgr.dev
    timestamp-authority: https://tsa.enforce.dev
```

For more guidance on working with the `chainctl` config, review our guide on [How to Manage `chainctl` Configuration](/chainguard/chainguard-enforce/manage-chainctl-config/).

## Policies 

> If you're running into issues with policies or cluster image policies, this section has some tips for recovery.

Reviewer: ```suggestion
If you're running into issues with policies, this section has some tips for recovery.
```
I don't think they're different enough to present them as separate ideas here. Like, I think I could articulate the difference between "policies" and "cluster image policies" but I'm not sure it matters to users

Author: Ok! I was trying to be specific but I agree

### Policy does not cover a given image as expected

When working with the `ClusterImagePolicy`, note that the glob wildcard `*` does not cover the `/` character. For example, a glob pattern like `gcr.io/apple/*` will cover paths like `gcr.io/apple/yak` or `gcr.io/apple/zebra`, but not `gcr.io/apple/banana/cashew`.

To match _everything_ — including the `/` character — use the `**` wildcard instead.

### Check that the cluster image policy is in the ConfigMap

You can ensure that a cluster image policy is in the [ConfigMap](https://kubernetes.io/docs/concepts/configuration/configmap/) with the following `kubectl` command:

```sh
kubectl get cm config-image-policies -n cosign-system 
```

> A describe of the ConfigMap will show the full policy. 

Reviewer: ```suggestion
A `describe` of the ConfigMap will show the full policy. 
```

> You can create a policy covering your image [through the Chainguard Enforce console](/chainguard/chainguard-enforce/policies/chainguard-policies-ui/) or [using the `chainctl` command line tool](/chainguard/chainguard-enforce/policies/chainguard-policies-cli/). Additionally, you can check out our [policy catalog](https://console.enforce.dev/policies/catalog) for our collection of policies that work directly with Chainguard Enforce.

Reviewer: ```suggestion
### I don't have any SBOMs

We recently updated Chainguard Enforce so it will generate SBOMs for any container images it finds that don't already have one. It can now also ingest SBOMs automatically out of the box.

Check out our guide on [Generating and Filtering SBOMs with Chainguard Enforce](/chainguard/chainguard-enforce/sboms/sboms-in-enforce/) to learn more about these features.
```
Also delete previous paragraph and h3
